src/serac/physics/thermal_conduction_functional.hpp

```diff
@@ -384,6 +487,79 @@ class ThermalConductionFunctional : public BasePhysics {
     }
   }
 
+  /**
+   * @brief Solve the adjoint problem
+   * @pre It is expected that the forward analysis is complete and the current temperature state is valid
+   * @note If the essential boundary state is not specified, homogeneous essential boundary conditions are applied
+   *
+   * @param[in] adjoint_load The dual state that contains the right hand side of the adjoint system
+   * @param[in] dual_with_essential_boundary A optional finite element dual containing the non-homogenous essential
+   * boundary condition data for the adjoint problem
+   * @return The computed adjoint finite element state
+   */
+  virtual const serac::FiniteElementState& solveAdjoint(FiniteElementDual& adjoint_load,
+                                                        FiniteElementDual* dual_with_essential_boundary = nullptr)
+  {
+    adjoint_load.initializeTrueVec();
+
+    // note: The assignment operator must be called after the copy constructor because
+    // the copy constructor only sets the partitioning, it does not copy the actual vector
+    // values
+    mfem::HypreParVector adjoint_load_vector(adjoint_load.trueVec());
+    adjoint_load_vector = adjoint_load.trueVec();
```

Comment:
Is the second assignment necessary?

Reply:
I'll check - but I believe yes. The copy constructor only copied the partitioning and not the data (or at least did at one point).

Reply:
No - this is no longer needed. I will remove this.